### PR TITLE
oce: fix for Sierra

### PR DESCRIPTION
**Pull request by davydden**
tested on `10.12` with `clang@8.0`.

see https://github.com/tpaviot/oce/issues/643

### CODE REVIEW

File: var/spack/repos/builtin/packages/oce/sierra.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+diff --git a/src/OSD/OSD_Chronometer.cxx b/src/OSD/OSD_Chronometer.cxx
+index 444f844..25970d0 100644
+--- a/src/OSD/OSD_Chronometer.cxx
++++ b/src/OSD/OSD_Chronometer.cxx
+@@ -51,7 +51,7 @@
+   #include <mach/mach.h>
+ #endif
+
+-#if defined(__APPLE__) && defined(__MACH__)
++#if defined(__OOOOPPLE__) && defined(__MACH__)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/oce/sierra.patch]: yeah, it is quick and dirty way to disable this, essentially...

**Comment by ilovezfs** [comment]
@davydden there was a bit lost in translation here going from the Homebrew patch to this fix. 

TLDR: the patch is also required on El Capitan if you have Xcode 8 installed.

Full story:

The Homebrew patch applies `if DevelopmentTools.clang_version >= "8.0"`, which is not an equivalent condition to `platform.mac_ver()[0] == '10.12'`.

Homebrew sets `DevelopmentTools.clang_version` equal to `8.0` in the following circumstances:

On El Capitan:
1) Xcode 8 is installed and CLT 7 is installed
2) Xcode 8 is installed and CLT 7 is not installed

or in other words, if Xcode 8 is installed regardless of the status of the CLT.
Note there is no CLT 8 for EL Captian

On Sierra:
1) Xcode 8 is installed and CLT 8 is installed
2) Xcode 8 is installed and CLT 8 is not installed
3) Xcode 8 is not installed and CLT 8 is installed

or in other words, if Xcode 8 is installed or CLT 8 is installed or both

It does not set DevelopmentTools.clang_version equal to `8.0` in the following circumstances::
On El Capitan:
1) Xcode 8 is not installed and Xcode 7 is installed and CLT 7 is installed
2) Xcode 8 is not installed and Xcode 7 is installed and CLT 7 is not installed
3) Xcode 8 is not installed and Xcode 7 is not installed and CLT 7 is installed

On Sierra:
Never unless the system is misconfigured in some way (Xcode 7 does not work on Sierra).

**Comment by davydden** [comment]
@ilovezfs thanks for the heads up, i still hope for the proper solution upstream.
